src/app/shared/addressHelper.ts

```diff
@@ -61,19 +58,19 @@ export const generateHTMLofAddress = (address: PostalAddressClass): string => {
 
   const fullBuildingName = buildingNames.join(", ");
   let fullStreetName;
-  if (streetNames) {
-    fullStreetName = streetNames.join(" ");
+  if (streetNames.length > 0) {
+    fullStreetName = streetNames.join(", ");
   }
 
-  const fullLocality = localityNames.join(" ");
+  const fullLocality = localityNames.join(", ");
 
-  if (fullStreetName) {
-    if (fullBuildingName) {
-      return `${fullBuildingName}<br>${fullStreetName}<br>${fullLocality}<br>${address.postalCode}`;
-    } else {
-      return `${fullStreetName}<br>${fullLocality}<br>${address.postalCode}`;
-    }
-  } else {
-    return `${fullBuildingName}<br>${fullLocality}<br>${address.postalCode}`;
-  }
+  let html = "";
+  if (fullBuildingName) html += `${fullBuildingName}<br>`;
+  if (fullStreetName) html += `${fullStreetName}<br>`;
+  if (fullLocality) html += `${fullLocality}<br>`;
+  if (address.addressRegion) html += `${address.addressRegion}<br>`;
+  html += `${address.postalCode}`;
+  if (address.addressCountry) html += `<br>${address.addressCountry}`;
```

Comment:
My understanding from the ticket is we shouldn't include country if it's a GB address?
Also I'm unsure about including this even for intl' addresses given it's a two-letter ISO code 🤔 I'll raise with JoeU